File: app/src/main/java/com/test/kabak/openweather/core/viewModels/AddCityViewModel.java
package com.test.kabak.openweather.core.viewModels;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.test.kabak.openweather.core.Resource;
import com.test.kabak.openweather.core.repositories.AddCityRepository;
import com.test.kabak.openweather.core.storage.City;

import java.util.List;

import io.reactivex.Completable;

public class AddCityViewModel extends ViewModel {
    AddCityRepository addCityRepository = new AddCityRepository();
    LiveData<Resource<List<City>>> searchLiveData;

    public LiveData<Resource<List<City>>> setSearchLiveData() {
        if(searchLiveData == null) {
            searchLiveData = addCityRepository.getSearchLiveData();
        }

        return searchLiveData;
    }

    public void findCities(String searchString) {
        addCityRepository.findCities(searchString);
    }

    public Completable addCity(City city) {
        return addCityRepository.saveCity(city);
    }
}

File: app/src/main/java/com/test/kabak/openweather/ui/addCity/AddCityViewModel.kt
package com.test.kabak.openweather.ui.addCity

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.test.kabak.openweather.core.Resource
import com.test.kabak.openweather.core.repositories.AddCityRepository
import com.test.kabak.openweather.data.db.Storage
import com.test.kabak.openweather.data.db.entity.CityTable
import kotlinx.coroutines.launch
import org.koin.java.KoinJavaComponent.inject

class AddCityViewModel : ViewModel() {
    private val storage: Storage by inject(Storage::class.java)

    var addCityRepository = AddCityRepository()
    var searchLiveData: LiveData<Resource<List<CityTable>>>? = null

    fun setSearchLiveData(): LiveData<Resource<List<CityTable>>> {
        if (searchLiveData == null) {
            searchLiveData = addCityRepository.searchLiveData
        }

        return searchLiveData!!
    }

    fun findCities(searchString: String?) {
        addCityRepository.findCities(searchString)
    }

    fun addCity(cityTable: CityTable) {
        viewModelScope.launch {
            storage.saveCity(cityTable)
        }
    }
}